### PR TITLE
Improve comments / Cleanup

### PR DESCRIPTION
**Pull request by FauziAkram**
Improve comments/ Cleanup
Tests used to derive some Elo worth comments:

https://tests.stockfishchess.org/tests/view/653cf6b7cc309ae83956263a
https://tests.stockfishchess.org/tests/view/655250b7136acbc573534711
https://tests.stockfishchess.org/tests/view/65525767136acbc5735347b9
https://tests.stockfishchess.org/tests/view/65525aa1136acbc573534801

bench: 1334248
no functional change

### CODE REVIEW

File: src/misc.cpp [hunk 1]
```diff
@@ -660,8 +660,7 @@ static int best_node(size_t idx) {
             groups.push_back(n);
 
     // In case a core has more than one logical processor (we assume 2) and we
-    // have still threads to allocate, then spread them evenly across available
-    // nodes.
+    // have still threads to allocate, spread them evenly across available nodes.
```
Comment (Disservin) [inline src/misc.cpp:663]: `still have`

File: src/movepick.h [hunk 2]
```diff
@@ -99,8 +99,8 @@ enum StatsType {
 // ButterflyHistory records how often quiet moves have been successful or
 // unsuccessful during the current search, and is used for reduction and move
 // ordering decisions. It uses 2 tables (one for each color) indexed by
-// the move's from and to squares, see www.chessprogramming.org/Butterfly_Boards
-// (~11 elo)
+// the move's from and to squares, as well as whether the to square is attacked,
```
Comment (Disservin) [inline src/movepick.h:102]: ` as well as whether the to square is attacked, ` I dont get this part?

File: src/search.cpp [hunk 3]
```diff
@@ -1246,6 +1245,8 @@ Value search(Position& pos, Stack* ss, Value alpha, Value beta, Depth depth, boo
             RootMove& rm =
               *std::find(thisThread->rootMoves.begin(), thisThread->rootMoves.end(), move);
 
+            // Update root move average score, using a weighted average with the new score value.
+            // Initialize to value if unset.
```
Comment (Disservin) [inline src/search.cpp:1249]: unnecessary imo

**Review by Disservin — CHANGES_REQUESTED** [review]

File: src/movepick.h (hunk 2 above)
Comment (FauziAkram) [reply src/movepick.h:102]: Removed

File: src/search.cpp (hunk 3 above)
Comment (FauziAkram) [reply src/search.cpp]: Removed

**Review by Disservin — APPROVED** [review]

**Pull request closed without merging**